src/Microsoft.AspNet.OData/Extensions/HttpRequestMessageProperties.cs

```diff
@@ -63,6 +65,48 @@ internal Func<long> TotalCountFunc
             }
         }
 
+        /// <summary>
+        /// Page size to be used by skiptoken implementation for the top-level resource for the request. 
+        /// </summary>
+        internal int PageSize
+        {
+            get
+            {
+                object pageSize;
+                if (_request.Properties.TryGetValue(PageSizeKey, out pageSize))
+                {
+                    return (int)pageSize;
+                }
+                return -1;
+            }
+            set
+            {
+                _request.Properties[PageSizeKey] = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the parsed OData <see cref="ODataQueryOptions"/> of the request. The
+        /// <see cref="ODataResourceSet "/> will use this information (if any) while writing the response for
+        /// this request.
+        /// </summary>
+        internal ODataQueryOptions QueryOptions
```

Comment:
>internal [](start = 8, length = 8)

We should deprecate SelectExpandClause in favor of QueryOptions, and in 8.x remove SelectExpandClause so we don't keep duplicate info.